bundles/org.openhab.binding.anthem/src/main/java/org/openhab/binding/anthem/internal/handler/AnthemHandler.java

```diff
@@ -0,0 +1,440 @@
+            localPollingJob.cancel(true);
+            this.pollingJob = null;
+        }
+
+        ScheduledFuture<?> localConnectRetryJob = this.connectRetryJob;
+        if (localConnectRetryJob != null) {
+            localConnectRetryJob.cancel(true);
+            this.connectRetryJob = null;
+        }
+
+        Thread localSenderThread = this.senderThread;
+        if (localSenderThread != null && localSenderThread.isAlive()) {
+            localSenderThread.interrupt();
+        }
+
+        Thread localReaderThread = this.readerThread;
+        if (localReaderThread != null && localReaderThread.isAlive()) {
+            localReaderThread.interrupt();
+        }
+        Socket localSocket = this.socket;
+        if (localSocket != null) {
+            try {
+                localSocket.close();
+            } catch (IOException e) {
+                logger.debug("Error closing socket: {}", e.getMessage());
+            }
+            this.socket = null;
+        }
+        BufferedReader localReader = this.reader;
+        if (localReader != null) {
+            try {
+                localReader.close();
+            } catch (IOException e) {
+                logger.debug("Error closing reader: {}", e.getMessage());
+            }
+            this.reader = null;
+        }
+        BufferedWriter localWriter = this.writer;
+        if (localWriter != null) {
+            try {
+                localWriter.close();
+            } catch (IOException e) {
+                logger.debug("Error closing writer: {}", e.getMessage());
+            }
+            this.writer = null;
+        }
+    }
+
+    private synchronized void reconnect() {
+        logger.debug("Attempting to reconnect to the Anthem");
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                "@text/thing-status-detail-reconnecting");
+        disconnect();
+        connect();
+    }
+
+    private void senderThreadJob() {
+        logger.debug("Sender thread started");
+        try {
+            while (!Thread.currentThread().isInterrupted() && writer != null) {
+                AnthemCommand command = sendQueue.take();
+                logger.debug("Sender thread writing command: {}", command);
+                try {
+                    BufferedWriter localWriter = this.writer;
+                    if (localWriter != null) {
+                        localWriter.write(command.toString());
+                        localWriter.flush();
+                    }
+                } catch (InterruptedIOException e) {
+                    logger.debug("Interrupted while sending command");
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "@text/thing-status-detail-interrupted");
+                    break;
+                } catch (IOException e) {
+                    logger.warn("Communication error, will try to reconnect. Error: {}", e.getMessage());
```

Comment:
Status change below is already logged, so this seems unneeded. You could add a third parameter with a description in the `updateStatus()` call. At the very least, this log should be reduced to **debug** if you don't want the exception text shown in the status.

Reply:
Reduced to debug